HW17/src/ThreadSafeList.java

```diff
@@ -0,0 +1,131 @@
+import java.util.Collection;
+import java.util.Iterator;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.ListIterator;
+
+public class ThreadSafeList<E> implements List<E> {
+
+    private final List<E> threadSafeList = new LinkedList<>();
+
+    @Override
+    public int size() {
+        return 0;
+    }
+
+    @Override
+    public boolean isEmpty() {
+        return false;
+    }
+
+    @Override
+    public boolean contains(Object o) {
+        return false;
+    }
+
+    @Override
+    public Iterator iterator() {
+        return null;
+    }
+
+    @Override
+    public Object[] toArray() {
+        return new Object[0];
+    }
+
+    @Override
+    public Object[] toArray(Object[] a) {
+        return new Object[0];
+    }
+
+    @Override
+    public boolean add(Object o) {
+        synchronized (threadSafeList) {
```

Comment:
I recommend using a standalone object as a lock. Good practice from the perspective of responsibility segregation (one thing = one functionality).

`private final Object lock = new Object()`